server/sandbox_status.go

```diff
@@ -55,6 +57,25 @@ func (s *Server) PodSandboxStatus(ctx context.Context, req *pb.PodSandboxStatusR
 		},
 	}
 
+	if req.Verbose {
+		resp = amendVerboseInfo(resp)
+	}
+
 	logrus.Debugf("PodSandboxStatusResponse: %+v", resp)
 	return resp, nil
 }
+
+// VersionPayload is a helper struct to create the JSON payload to show the version
+type VersionPayload struct {
+	Version string `json:"version"`
+}
+
+func amendVerboseInfo(resp *pb.PodSandboxStatusResponse) *pb.PodSandboxStatusResponse {
+	resp.Info = make(map[string]string)
+	bs, err := json.Marshal(VersionPayload{Version: version.Version})
+	if err != nil {
+		return resp // Just ignore the error and don't marshal the info
+	}
+	resp.Info["version"] = string(bs)
```

Comment:
Do we need the `VersionPayload` helper?  Can't we skip `bs` and use:

```golang
resp.Info["version"] = version.Version
```

Reply:
The [documentation says that it should be in JSON format](https://github.com/kubernetes/kubernetes/blob/c6ddc749e85f4281628f366d65cf924c22d6cd99/pkg/kubelet/apis/cri/v1alpha1/runtime/api.proto#L387-L391), however, a string is considered valid JSON.

To be fair I am not fully sure what's expected there and I couldn't find any example from other implementation. Since I was adding `version` as a placeholder for more useful information I didn't give it that much thought for now.

Reply:
> Since I was adding `version` as a placeholder for more useful information I didn't give it that much thought for now.

But that extension will be adding additional siblings in `resp.Info[…]`, right?  Not adding siblings within `VersionPayload`?